### PR TITLE
Turned off a test on Linux that was failing because file permissions …

### PR DESCRIPTION
**Pull request by AMZN-stankowi**
…in our Python libraries don't work on Linux.

Updated the lock/unlock file functions to emit a log and not a warning when they do what they are called to do.

Signed-off-by: AMZN-stankowi <4838196+AMZN-stankowi@users.noreply.github.com>

## What does this PR do?

Disables a test on Linux that does not function on Linux.

## How was this PR tested?

Tested on Windows and Linux, test now passes on Linux, continues to pass on Windows.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_relocator_tests.py [hunk 1]
```diff
@@ -344,6 +344,10 @@ def test_WindowsMacPlatforms_DeleteWithParameters_DeleteSuccess(
         4. Assert file existence or nonexistence based on the test case
         5. Validate the relocation report based on expected and unexpected messages
         """
+        if str(type(workspace)) == "<class 'ly_test_tools._internal.managers.platforms.linux.LinuxWorkspaceManager'>":
+            logger.info("Skipping test on Linux, Python based file locking does not function on Linux.")
+            return
```
Comment (smurly) [inline AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_relocator_tests.py:349]: won't this test execute and log as a passing test on linux? that accomplishes your goal of not failing on linux but I'm not sure that's the best way to do this.

Comment (AMZN-stankowi) [reply AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_relocator_tests.py:349]: Yeah, I'd rather have the test be marked skipped on Linux, but I couldn't find a clear way to conditionally skip tests that way.

Comment (Kadino) [reply AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_relocator_tests.py:349]: You can annotate the test:
```python
@pytest.mark.skipif(ly_test_tools.LINUX, reason="please mention why!")
```
or raise a skip-exception from inside the test:
```python
if ly_test_tools.LINUX:
    pytest.skip(reason="please mention why!")
```

Comment (smurly) [reply AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_relocator_tests.py:349]: I've seen CMakeLists.txt used if you look in .\o3de\AutomatedTesting\Gem\PythonTests\assetpipeline\asset_processor_tests\CMakeLists.txt line 150 there's a section you can place a suite of tests that are not to be run on linux. after that check I believe the 3 suites below there are windows mac only. so your test could go into a suite there or you could add a suite for windows mac only that resides below that platform check

Comment (AMZN-stankowi) [reply AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_relocator_tests.py]: Thanks all, that works. I've updated the test, verified it's skipping and not causing all test execution to stop on Linux, and verified it still runs on Windows.

**Review by smurly — APPROVED** [review]

**Pull request merged**